libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_mqtt_transport.c

```diff
@@ -41,20 +41,73 @@
  */
 #define SIZE_OF_SUB_ACK       5U
 
+#define MAX_ENCODE_LENGTH     4
+
+#define BIT_MASK( bitnum )    ( ( uint8_t ) ( 1 << bitnum ) )
 
 /* Define masks for each flag in Connect packets */
-#define CLEAN_SESSION_MASK    0x02U
+#define CLEAN_SESSION_MASK                0x02U
+
+#define WILL_FLAG_MASK                    0X04U
 
-#define WILL_FLAG_MASK        0X04U
+#define WILL_QOS_MASK                     0x18U
 
-#define WILL_QOS_MASK         0x18U
+#define WILL_RETAIN_MASK                  0x20U
 
-#define WILL_RETAIN_MASK      0x20U
+#define PASSWORD_MASK                     0x40U
 
-#define PASSWORD_MASK         0x40U
+#define USERNAME_MASK                     0x80U
 
-#define USERNAME_MASK         0x80U
+/**
+ * @brief Defines for the control packet type of the MQTT packets sent from the client to server.
+ */
+#define CLIENT_PACKET_TYPE_CONNECT        ( ( uint8_t ) 0x10U )  /**< @brief CONNECT (client-to-server). */
+#define CLIENT_PACKET_TYPE_PUBLISH        ( ( uint8_t ) 0x30U )  /**< @brief PUBLISH (bidirectional). */
+#define CLIENT_PACKET_TYPE_PUBACK         ( ( uint8_t ) 0x40U )  /**< @brief PUBACK (bidirectional). */
+#define CLIENT_PACKET_TYPE_PUBREC         ( ( uint8_t ) 0x50U )  /**< @brief PUBREC (bidirectional). */
+#define CLIENT_PACKET_TYPE_PUBREL         ( ( uint8_t ) 0x60U )  /**< @brief PUBREL (bidirectional). */
+#define CLIENT_PACKET_TYPE_PUBCOMP        ( ( uint8_t ) 0x70U )  /**< @brief PUBCOMP (bidirectional). */
+#define CLIENT_PACKET_TYPE_SUBSCRIBE      ( ( uint8_t ) 0x80U )  /**< @brief SUBSCRIBE (client-to-server). */
+#define CLIENT_PACKET_TYPE_UNSUBSCRIBE    ( ( uint8_t ) 0xA0U )  /**< @brief UNSUBSCRIBE (client-to-server). */
+#define CLIENT_PACKET_TYPE_PINGREQ        ( ( uint8_t ) 0xC0U )  /**< @brief PINGREQ (client-to-server). */
+#define CLIENT_PACKET_TYPE_DISCONNECT     ( ( uint8_t ) 0xE0U )  /**< @brief DISCONNECT (client-to-server). */
```

Comment:
Aren't this macros already defined in  the included header core_mqtt_serializer.h?

Reply:
No, these macros are used to deserialize the outgoing packet type and not used by `core_mqtt_serializer.h`.

Reply:
Aren't these the same https://github.com/FreeRTOS/coreMQTT/blob/a867f4cf4819f3bc9a950e0f611e1db5f8ec7c42/source/include/core_mqtt_serializer.h#L71-L84?

Reply:
Some of the packet type macros are different for example `MQTT_PACKET_TYPE_UNSUBSCRIBE` is `0xA2U` as it includes the flag. The macros are used for serialization, whereas these macros includes only the packet type part.